### PR TITLE
Fixed andFlush parameter in MediaManager never being used

### PR DESCRIPTION
**Pull request by Uriziel**
Rebased changes of  PR #537 
Added tests compared to PR #796

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
You could add a PR on @prashantpalikhe repo to complete it instead! ;-)

Anyway, will take a look later. Thanks.

**Comment by Uriziel** [comment]
So?

File: Document/MediaManager.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ public function save($entity, $andFlush = true)
             $entity->setProviderName(func_get_arg(2));
         }
 
-        if ($andFlush && is_bool($andFlush)) {
+        if (is_bool($andFlush)) {
```
Comment (Soullivaneuh) [inline Document/MediaManager.php]: Just asking: With this condition, `false` and `true` would pass as `$andFlush` is a boolean.

Would be more logic to have:

```php
is_bool($andFlush) && $andFlush
```

Ore more simple:

```php
true === $andFlush
```

Strict comparison will check both type and value.

What do you think?

Comment (Uriziel) [inline Document/MediaManager.php]: The whole point of this commit is to get rid of comparing this to true as you might want to pass false value to the save method.

Comment (Soullivaneuh) [inline Document/MediaManager.php]: I don't understand.

Your condition just check if it's a boolean, never-mind if it's `true` or `false`.

I don't think this is the wanted behavior.

Ping @rande

Comment (Uriziel) [inline Document/MediaManager.php]: How about you checkout this branch, run tests, checkout HEAD^, run tests again.
I've separated test and code commits so you can easily find out how much it actually helps.
Also read test code to understand why one might want to pass false there

Comment (dvapelnik) [inline Document/MediaManager.php]: I want convert one thousand articles with images. But I don't wont flush changes after each articles with media object. I want create media-object, save it w/o flushing. So I'm writing:
```php
// ... creating new media with image
$mediaManager->save($media, false);
$em->persist(); // may by it's excessive
// ... again
// ... again
// ... again
$em->flush();
```
But my `false` flag was ignored and all entities flushed after each saved media with article, authors, author pseudonyms etc. 
I think that this way is incorrect. The developer should be able to choose and this choice should be transparent

**Comment by OskarStark** [comment]
ping @rande

**Comment by OskarStark** [comment]
@Uriziel any news on this PR?

**Comment by Uriziel** [comment]
@OskarStark it was all green when I made it :neutral_face: 

Fun stuff is that it probably just needs rebasing, as new test engines were introduced later on.
Given that this change is already waiting since early 2014 I don't seem to care anymore, if you decide to merge this I can rebase it so new test engines pass.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**